packages/amplify-e2e-core/src/categories/lambda-function.ts

```diff
@@ -350,20 +351,34 @@ export const selectRuntime = (chain: ExecutionContext, runtime: FunctionRuntimes
   singleSelect(chain, runtimeName, runtimeChoices);
 };
 
-export const selectTemplate = (chain: ExecutionContext, functionTemplate: string, runtime: FunctionRuntimes) => {
+export const selectTemplate = (chain: ExecutionContext, functionTemplate: string, runtime: FunctionRuntimes, appSyncAuthType?: string) => {
   const templateChoices = getTemplateChoices(runtime);
   chain.wait('Choose the function template that you want to use');
 
   // reset cursor to top of list because Hello World is default but it throws off offset calculations
   moveUp(chain, templateChoices.indexOf('Hello World'));
 
   singleSelect(chain, functionTemplate, templateChoices);
+
+  if (functionTemplate === 'AppSync Todo' && appSyncAuthType === 'API_KEY') {
```

Comment:
`if (functionTemplate === 'AppSync Todo' && (!appSyncAuthType || appSyncAuthType === 'API_KEY'))`
and can remove the block under `just in case` this way you can handle all falsy values ('', undefined, null)